internal/middleware/reverse_proxy.go

```diff
@@ -29,6 +32,9 @@ func SignRequest(signer cryptutil.JWTSigner, id, email, groups, header string) f
 func StripPomeriumCookie(cookieName string) func(next http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, span := telemetry.StartSpan(r.Context(), "middleware.StripPomeriumCookie")
```

Comment:
Should this be keeping the returned context and calling r.WithContext() when we ServeHTTP on the next middleware?

Reply:
Yes, very much so. great catch.